src/d3a/d3a_core/live_events.py

```diff
@@ -53,6 +53,9 @@ def apply(self, area):
         self.sanitize_live_event_paramters_for_strategy_update()
         area_type = self.area_params.pop("type", None)
         if area_type is not None:
```

Comment:
I know it is a quickfix, but please refactor it so that `area.area_reconfigure_event(**self.area_params)` is always called (apart from the case when if area_type == "MarketMaker": or elif area_type == "InfiniteBus":

Reply:
The area_type should never be None, right?